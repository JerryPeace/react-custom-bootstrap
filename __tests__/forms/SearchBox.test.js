import React from 'react';
import {mount} from 'enzyme';
import SearchBox from 'forms/searchBox';

const defaultValue = 'defaultValue';

describe('SearchBox', () => {
    let searchBox;
    test('should render corrently', () => {
        searchBox = mount(<SearchBox defaultValue={defaultValue} />);
        expect(searchBox).toMatchSnapshot();
    });

    test('could show search button', () => {
        searchBox = mount(<SearchBox displaySearchBtn={true} />);
        expect(searchBox).toMatchSnapshot();
    });

    test('could hide search button', () => {
        searchBox = mount(<SearchBox displaySearchBtn={false} />);
        expect(searchBox).toMatchSnapshot();
    });

    describe('could call onSearch successfully', () => {
        let searchFn;
        beforeEach(() => {
            searchFn = jest.fn();
            searchBox = mount(<SearchBox displaySearchBtn={true} onSearch={searchFn} />);
        });
        test('by search button', () => {
            searchBox.find('button').simulate('click');
            expect(searchFn.mock.calls.length).toBe(1);
        });
        test('by typing "Enter"', () => {
            searchBox.find('input').simulate('keyUp', {keyCode: 13});
            expect(searchFn.mock.calls.length).toBe(1);
        });

    });
});
