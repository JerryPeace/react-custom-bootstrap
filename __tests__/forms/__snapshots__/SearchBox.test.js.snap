// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox could hide search button 1`] = `
<SearchBox
  displaySearchBtn={false}
  onSearch={[Function]}
  searchBtnProps={Object {}}
  textFieldProps={Object {}}
>
  <InputGroup
    bsClass="input-group"
    className="search-box"
    inputFieldRenderer={[Function]}
    rightField={null}
    rightFieldType="button-group"
  >
    <div
      className="search-box input-group"
    >
      <ClearableTextField
        inputType="text"
        leftIconName="search-o"
        onClear={[Function]}
        onKeyUp={[Function]}
        showLeftIcon={false}
      >
        <div
          className="search-box-input"
        >
          <Textfield
            bsClass="input"
            delayValidateTime={500}
            disabled={false}
            helpText=""
            helpTextClassName="has-error"
            inputType="text"
            labelAlign="left"
            labelClassName="col-xs-2"
            layoutHelperClassName=""
            multiline={false}
            onChange={[Function]}
            onKeyUp={[Function]}
            standalone={true}
            validator={[Function]}
            wrapperClassName="col-xs-10"
          >
            <LayoutHelper
              FieldBase={
                <input
                  className=""
                  disabled={false}
                  layoutHelperClassName=""
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  type="text"
                />
              }
              bsClass="input"
              delayValidateTime={500}
              disabled={false}
              getInputEl={[Function]}
              helpText={null}
              helpTextClassName="has-error"
              inputType="text"
              labelAlign="left"
              labelClassName="col-xs-2"
              layoutHelperClassName=""
              multiline={false}
              onChange={[Function]}
              onKeyUp={[Function]}
              singleCheckboxRadio={false}
              standalone={true}
              validator={[Function]}
              wrapperClassName="col-xs-10"
            >
              <div
                className=""
              >
                <input
                  className="form-control"
                  disabled={false}
                  layoutHelperClassName=""
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  type="text"
                />
              </div>
            </LayoutHelper>
          </Textfield>
        </div>
      </ClearableTextField>
    </div>
  </InputGroup>
</SearchBox>
`;

exports[`SearchBox could show search button 1`] = `
<SearchBox
  displaySearchBtn={true}
  onSearch={[Function]}
  searchBtnProps={Object {}}
  textFieldProps={Object {}}
>
  <InputGroup
    bsClass="input-group"
    className="search-box has-search-btn"
    inputFieldRenderer={[Function]}
    rightField={
      <Button
        active={false}
        block={false}
        bsClass="button"
        bsStyle="default"
        className="serach-box-btn"
        disabled={false}
        faProps={Object {}}
        faStyle="search"
        isCaretStyle={false}
        isDropdown={false}
        isIconOnly={true}
        isLoading={false}
        onClick={[Function]}
        standalone={true}
        type="button"
      />
    }
    rightFieldType="button-group"
  >
    <div
      className="search-box has-search-btn input-group"
    >
      <ClearableTextField
        inputType="text"
        leftIconName="search-o"
        onClear={[Function]}
        onKeyUp={[Function]}
        showLeftIcon={false}
      >
        <div
          className="search-box-input"
        >
          <Textfield
            bsClass="input"
            delayValidateTime={500}
            disabled={false}
            helpText=""
            helpTextClassName="has-error"
            inputType="text"
            labelAlign="left"
            labelClassName="col-xs-2"
            layoutHelperClassName=""
            multiline={false}
            onChange={[Function]}
            onKeyUp={[Function]}
            standalone={true}
            validator={[Function]}
            wrapperClassName="col-xs-10"
          >
            <LayoutHelper
              FieldBase={
                <input
                  className=""
                  disabled={false}
                  layoutHelperClassName=""
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  type="text"
                />
              }
              bsClass="input"
              delayValidateTime={500}
              disabled={false}
              getInputEl={[Function]}
              helpText={null}
              helpTextClassName="has-error"
              inputType="text"
              labelAlign="left"
              labelClassName="col-xs-2"
              layoutHelperClassName=""
              multiline={false}
              onChange={[Function]}
              onKeyUp={[Function]}
              singleCheckboxRadio={false}
              standalone={true}
              validator={[Function]}
              wrapperClassName="col-xs-10"
            >
              <div
                className=""
              >
                <input
                  className="form-control"
                  disabled={false}
                  layoutHelperClassName=""
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  type="text"
                />
              </div>
            </LayoutHelper>
          </Textfield>
        </div>
      </ClearableTextField>
      <span
        className="input-group-btn"
      >
        <Button
          active={false}
          block={false}
          bsClass="button"
          bsStyle="default"
          className="serach-box-btn"
          disabled={false}
          faProps={Object {}}
          faStyle="search"
          isCaretStyle={false}
          isDropdown={false}
          isIconOnly={true}
          isLoading={false}
          onClick={[Function]}
          standalone={true}
          type="button"
        >
          <button
            className="btn btn-default btn-icon-only serach-box-btn"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="btn-icon"
            >
              <i
                className="fa fa-search"
              />
            </span>
          </button>
        </Button>
      </span>
    </div>
  </InputGroup>
</SearchBox>
`;

exports[`SearchBox should render corrently 1`] = `
<SearchBox
  defaultValue="defaultValue"
  displaySearchBtn={false}
  onSearch={[Function]}
  searchBtnProps={Object {}}
  textFieldProps={Object {}}
>
  <InputGroup
    bsClass="input-group"
    className="search-box"
    inputFieldRenderer={[Function]}
    rightField={null}
    rightFieldType="button-group"
  >
    <div
      className="search-box input-group"
    >
      <ClearableTextField
        inputType="text"
        leftIconName="search-o"
        onClear={[Function]}
        onKeyUp={[Function]}
        showLeftIcon={false}
      >
        <div
          className="search-box-input"
        >
          <Textfield
            bsClass="input"
            delayValidateTime={500}
            disabled={false}
            helpText=""
            helpTextClassName="has-error"
            inputType="text"
            labelAlign="left"
            labelClassName="col-xs-2"
            layoutHelperClassName=""
            multiline={false}
            onChange={[Function]}
            onKeyUp={[Function]}
            standalone={true}
            validator={[Function]}
            wrapperClassName="col-xs-10"
          >
            <LayoutHelper
              FieldBase={
                <input
                  className=""
                  disabled={false}
                  layoutHelperClassName=""
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  type="text"
                />
              }
              bsClass="input"
              delayValidateTime={500}
              disabled={false}
              getInputEl={[Function]}
              helpText={null}
              helpTextClassName="has-error"
              inputType="text"
              labelAlign="left"
              labelClassName="col-xs-2"
              layoutHelperClassName=""
              multiline={false}
              onChange={[Function]}
              onKeyUp={[Function]}
              singleCheckboxRadio={false}
              standalone={true}
              validator={[Function]}
              wrapperClassName="col-xs-10"
            >
              <div
                className=""
              >
                <input
                  className="form-control"
                  disabled={false}
                  layoutHelperClassName=""
                  onChange={[Function]}
                  onKeyUp={[Function]}
                  type="text"
                />
              </div>
            </LayoutHelper>
          </Textfield>
        </div>
      </ClearableTextField>
    </div>
  </InputGroup>
</SearchBox>
`;
